flutter/Graminsta/lib/auth/login.dart

```diff
@@ -1,11 +1,46 @@
 import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:Graminsta/http.dart';
+import 'package:dio/dio.dart';
 
 class LoginPage extends StatefulWidget {
   @override
   _LoginPageState createState() => new _LoginPageState();
 }
 
 class _LoginPageState extends State<LoginPage> {
+  final _usernameController = TextEditingController();
+  final _passwordController = TextEditingController();
+
+  void login() async {
+    Response response;
+    HttpManager.post("core/login/", data: {
```

Comment:
Could we have another layer of abstraction like `auth_service.dart`. And maybe we can use [http](https://pub.dev/packages/http) internally instead of `HttpManager`, since it's easy and enough for our case?

```
// http_service.dart
// Class that handles HTTP requests.
class _HttpService {
  static _HttpService _instance;

  final globalHeaders = Map<String, string>();

  factory _HttpService() {
    _instance ??= _HttpService._();
    return _instance;
  }

  _HttpService._() {}

  // Set the auth token for authentication.
  Future<void> setAuthToken(String token);
  
  Future<Response> get(String url, {Map<String, String> headers});

  Future<Response> post(String url, {Map<String, String> headers, Object body, Encoding encoding});
  
  ...
}

// Variable to allow other services to use _HttpService easily.
final http = _HttpService();

// auth_service.dart

import 'path/to/http_service.dart';

class AuthService {
  static AuthService _instance;

  factory AuthService() {
    _instance ??= AuthService._();
    return _instance;
  }

  AuthService._() {}

  Future<void> logIn(String username, String password) {
    http.post(xxx)
  }
}
```

See also: https://flutter.dev/docs/cookbook/networking/fetch-data
https://pub.dev/documentation/http/latest/http/http-library.html